frontend/public/components/dashboard/details-card/detail-item.tsx

```diff
@@ -1,19 +1,28 @@
 import * as React from 'react';
-import classNames from 'classnames';
-
 import { LoadingInline } from '../../utils';
 
-export const DetailItem: React.FC<DetailItemProps> = React.memo(({ title, value, isLoading }) => (
-  <React.Fragment>
-    <dt className="co-details-card__item-title">{title}</dt>
-    <dd className={classNames('co-details-card__item-value', {'text-secondary': !value})}>
-      {isLoading ? <LoadingInline /> : value || 'Unavailable'}
-    </dd>
-  </React.Fragment>
-));
+export const DetailItem: React.FC<DetailItemProps> = React.memo(
+  ({ title, isLoading, children, error = false }) => {
+    let status: React.ReactNode;
+    if (isLoading) {
+      status = <LoadingInline />;
+    } else if (error) {
+      status = <span className="text-secondary">Unavailable</span>;
+    } else {
+      status = children;
+    }
+    return (
+      <React.Fragment>
+        <dt className="co-details-card__item-title">{title}</dt>
+        <dd className="co-details-card__item-value">{status}</dd>
+      </React.Fragment>
+    );
+  }
+);
 
 type DetailItemProps = {
   title: string;
-  value?: string;
   isLoading: boolean;
+  children: React.ReactNode;
```

Comment:
FYI there is a React helper type to avoid repeating `children` in your component props:

```ts
// node_modules/@types/react/index.d.ts
type PropsWithChildren<P> = P & { children?: ReactNode };
```

You can use it like so:

```ts
type DetailItemProps = React.PropsWithChildren<{
  title: string;
  isLoading: boolean;
  error: boolean;
}>;
```

and it will add an optional `children` prop.

Reply:
nice!

Reply:
Good to know

Reply:
ack

Reply:
React.FC already includes optional children prop as it uses that helper internally, so it is not necessary to deal with children prop at all (except for the cases when we want to explicitly define children prop as specific type or make it required)

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L495